src/main/java/org/broadinstitute/hellbender/tools/coveragemodel/CoverageModelEMComputeBlock.java

```diff
@@ -0,0 +1,1332 @@
+        B_st("B_{st} (refer to the technical white paper)"),
+        loglike_unreg("Contribution of this block of targets to the model log likelihood (w/o Fourier regularizer)"),
+        loglike_reg("Contribution of this block of targets to the model log likelihood (w/ Fourier regularizer)");
+
+        public final String description;
+
+        CoverageModelICGCacheNode(final String description) {
+            this.description = description;
+        }
+    }
+
+    /**
+     * Immutable computable graph cache tags
+     */
+    public enum CoverageModelICGCacheTag {
+        E_STEP_D("Cache nodes to be updated for the E-step for read depth"),
+        E_STEP_C("Cache nodes to be updated for the E-step for copy ratio (or copy number)"),
+        E_STEP_Z("Cache nodes to be updated for the E-step for bias continuous latent variables"),
+        E_STEP_GAMMA("Cache nodes to be updated for the E-step for sample-specific unexplained variance of the log bias"),
+        M_STEP_M("Cache nodes to be updated for the M-step for mean log bias"),
+        M_STEP_W_REG("Cache nodes to be updated for the M-step for bias covariances (w/ regularization)"),
+        M_STEP_W_UNREG("Cache nodes to be updated for the M-step for bias covariances (w/o regularization)"),
+        M_STEP_PSI("Cache nodes to be updated for the M-step for target-specific unexplained variance"),
+        LOGLIKE_UNREG("Cache nodes to be updated for log likelihood calculation (w/o regularization)"),
+        LOGLIKE_REG("Cache nodes to be updated for log likelihood calculation (w/ regularization)");
+
+        public final String description;
+
+        CoverageModelICGCacheTag(final String description) {
+            this.description = description;
+        }
+    }
+
+    /**
+     * This method creates an empty instance of {@link ImmutableComputableGraph} for worker-side
+     * calculations
+     *
+     * @return an instance of {@link ImmutableComputableGraph}
+     */
+    private static ImmutableComputableGraph createEmptyCacheGraph() {
+        final ImmutableComputableGraph.ImmutableComputableGraphBuilder cgbuilder =
+                ImmutableComputableGraph.builder();
+        /*
+         * Data nodes
+         */
+        cgbuilder
+                /* raw read counts */
+                .addPrimitiveNode(CoverageModelICGCacheNode.n_st.name(),
+                        new String[]{},
+                        new DuplicableNDArray())
+                /* mask */
+                .addPrimitiveNode(CoverageModelICGCacheNode.M_st.name(),
+                        new String[]{},
+                        new DuplicableNDArray())
+                /* mapping error rate */
+                .addPrimitiveNode(CoverageModelICGCacheNode.MER_st.name(),
+                        new String[]{},
+                        new DuplicableNDArray());
+
+        /*
+         * Model parameters
+         */
+        cgbuilder
```

Comment:
Good suggestion but the fact that ICGs for models w/ and w/o ARD and w/ and models w/o bias covariate are different (introduced in commit https://github.com/broadinstitute/gatk-protected/pull/857/commits/078ce453b599b138005176599afd91d36f961e6d) makes this approach not as concise and more mysterious-looking.